stackslib/src/net/api/blocksimulate.rs

```diff
@@ -0,0 +1,380 @@
+                let coinbase = block
+                    .txs
+                    .iter()
+                    .find(|tx| matches!(tx.payload, TransactionPayload::Coinbase(..)));
+                let tenure_cause = tenure_change.and_then(|tx| match &tx.payload {
+                    TransactionPayload::TenureChange(tc) => Some(tc.cause),
+                    _ => None,
+                });
+
+                let parent_stacks_header =
+                    NakamotoChainState::get_block_header(chainstate.db(), &parent_block_id)
+                        .unwrap()
+                        .unwrap();
+                let mut builder = NakamotoBlockBuilder::new(
+                    &parent_stacks_header,
+                    &block.header.consensus_hash,
+                    block.header.burn_spent,
+                    tenure_change,
+                    coinbase,
+                    block.header.pox_treatment.len(),
+                    None,
+                    None,
+                )
+                .unwrap();
+
+                let mut miner_tenure_info = builder
+                    .load_ephemeral_tenure_info(chainstate, &burn_dbconn, tenure_cause)
+                    .unwrap();
+                let burn_chain_height = miner_tenure_info.burn_tip_height;
+                let mut tenure_tx = builder
+                    .tenure_begin(&burn_dbconn, &mut miner_tenure_info)
+                    .unwrap();
+
+                let mut block_fees: u128 = 0;
+                let mut txs_receipts = vec![];
+
+                for (i, tx) in block.txs.iter().enumerate() {
+                    let tx_len = tx.tx_len();
+
+                    let tx_result = builder.try_mine_tx_with_len(
+                        &mut tenure_tx,
+                        tx,
+                        tx_len,
+                        &BlockLimitFunction::NO_LIMIT_HIT,
+                        None,
+                    );
+                    let err = match tx_result {
+                        TransactionResult::Success(tx_result) => {
+                            txs_receipts.push(tx_result.receipt);
+                            Ok(())
+                        }
+                        _ => Err(format!("Problematic tx {i}")),
+                    };
+                    if let Err(reason) = err {
+                        panic!("Rejected block tx: {reason}");
+                    }
+
+                    block_fees += tx.get_tx_fee() as u128;
+                }
+
+                let simulated_block =
+                    builder.mine_nakamoto_block(&mut tenure_tx, burn_chain_height);
+
+                tenure_tx.rollback_block();
+
+                let block_hash = block.header.block_hash();
+                let block_height = block.header.chain_length;
+
+                let tx_merkle_root = block.header.tx_merkle_root.clone();
+
+                let mut simulated_block = RPCSimulatedBlock {
```

Comment:
Considering that most of the fields in this struct are derived from `block`, can you please add a constructor function for `RPCSimulatedBlock` that takes the `block` as an argument?